internal/app/mapper/mapper.go

```diff
@@ -34,5 +38,44 @@ func (mapper *mapper) GetKey(request v2.DiscoveryRequest, typeURL string) (strin
 	if typeURL == "" {
 		return "", fmt.Errorf("typeURL is empty")
 	}
+
+	var resultFragments []string
+	for _, fragment := range mapper.config.GetFragments() {
+		fragmentrules := fragment.GetRules()
+		for _, fragmentrule := range fragmentrules {
+			matchpredicate := fragmentrule.GetMatch()
+			isMatch := isMatchPredicate(matchpredicate)
+			if isMatch {
+				result := getResult(fragmentrule)
+				resultFragments = append(resultFragments, result)
+			}
+		}
+	}
+
+	if len(resultFragments) != 0 {
+		return strings.Join(resultFragments, "_"), nil
+	}
+
 	return "", fmt.Errorf("Cannot map the input to a key")
 }
+
+func isMatchPredicate(matchPredicate *matchPredicate) bool {
+	if matchPredicate.GetAnyMatch() {
+		return isAnyMatch(matchPredicate)
+	}
+
+	return false
+}
+
+func isAnyMatch(matchPredicate *matchPredicate) bool {
```

Comment:
Would just kill this func and `return isAnyMatch(matchPredicate)` in the above func. It will be the final else condition when the other match predicate types are supported.

Reply:
makes sense.